apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.ts

```diff
@@ -145,21 +148,42 @@ router.get(
 router.post(
   '/:feature',
   asyncHandler(async (req, res) => {
-    const feature = validateFeature(req.params.feature);
+    if (req.body.__action === 'add_feature_grant') {
+      const feature = validateFeature(req.params.feature);
 
-    const params = AddFeatureGrantModalParamsSchema.parse(req.body);
-    const { institution, course, course_instance } = await getEntitiesFromParams(params);
+      const params = AddFeatureGrantModalParamsSchema.parse(req.body);
+      const { institution, course, course_instance } = await getEntitiesFromParams(params);
 
-    await features.enable(
-      feature,
-      features.validateContext({
+      const context = features.validateContext({
         institution_id: institution?.id ?? null,
         course_id: course?.id ?? null,
         course_instance_id: course_instance?.id ?? null,
-      }),
-    );
-
-    res.redirect(req.originalUrl);
+      });
+
+      if (params.enabled) {
+        await features.enable(feature, context);
+      } else {
+        await features.disable(feature, context);
+      }
+
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'toggle_feature_grant') {
```

Comment:
Again I think this is actually `update_enabled` rather than `toggle`, given that we pass the value of `enabled` that we want to set.